ui/app/src/modules/System/Publishing/Queue/PublishingQueue.tsx

```diff
@@ -204,42 +205,31 @@ function PublishingQueue(props: PublishingQueueProps) {
         setPackages(response.packages);
       },
       ({ response }) => {
-        setApiState({ ...apiState, error: true, errorResponse: response });
+        setApiState({ error: true, errorResponse: response });
       }
     )
-  ), [apiState, currentFilters]);
+  ), [currentFilters, setApiState]);
 
   setRequestForgeryToken();
 
   useEffect(() => {
+    fetchEnvironments(siteId).subscribe(
+      ({ response }) => {
+        let channels: string[] = [];
+        response.availablePublishChannels.forEach((channel: any) => {
+          channels.push(channel.name);
+        });
+        setFilters({ environments: channels });
+      },
+      ({ response }) => {
+        setApiState({ error: true, errorResponse: response });
+      }
+    );
+  }, [siteId, setFilters, setApiState]);
 
-    function getEnvironments(siteId: string) {
-      fetchEnvironments(siteId)
-        .subscribe(
-          ({ response }) => {
-            let channels: string[] = [];
-            response.availablePublishChannels.forEach((channel: any) => {
-              channels.push(channel.name);
-            });
-            setFilters({ ...filters, environments: channels });
-          },
-          ({ response }) => {
-            setApiState({ ...apiState, error: true, errorResponse: response });
-          }
-        );
-    }
-
-    if (currentFilters && packages !== null) {
-      getPackages(siteId);
-    }
-    if (packages === null) {
-      getPackages(siteId);
-    }
-    if (filters.environments === null) {
-      getEnvironments(siteId);
-    }
-
-  }, [apiState, currentFilters, filters, getPackages, packages, siteId]);
+  useEffect(() => {
+    getPackages(siteId);
+  }, [currentFilters, siteId, getPackages]);
```

Comment:
currentFilters is not a dependency of this effect

Reply:
it is not a dependency, but it needs to change everytime that the currentFilters changes